src/run/run.c

```diff
@@ -2116,6 +2117,7 @@ static int start_transient_service(sd_bus *bus) {
                         return r;
         }
 
+        _cleanup_(osc_context_closep) sd_id128_t osc_context_id = SD_ID128_NULL;
```

Comment:
I think this can be moved in the `if (arg_wait || arg_stdio != ARG_STDIO_NONE)` block below.